src/nvim/mark.c

```diff
@@ -68,6 +69,21 @@ int setmark(int c)
   return setmark_pos(c, &curwin->w_cursor, curbuf->b_fnum);
 }
 
+/// Free fmark_T item
+void free_fmark(fmark_T fm)
+{
+  dict_unref(fm.additional_data);
+  fm.additional_data = NULL;
```

Comment:
`fm` is a copy, so setting `fm.additional_data` changes nothing after this function returns. Should this function expect a `fmark_T*` argument instead?